src/Kernel/Magic/Simulate.cs

```diff
@@ -101,8 +101,12 @@ public async Task<ExecutionResult> RunAsync(string input, IChannel channel)
             var maxNQubits = 0L;
 
             using var qsim = new QuantumSimulator()
-                .WithJupyterDisplay(channel, ConfigurationSource)
+                //.WithJupyterDisplay(channel, ConfigurationSource)   // TODO(rokuzmin): Remove as a simplification of Dump().
                 .WithStackTraceDisplay(channel);
+            ////
+            qsim.DisableLogToConsole();
+            qsim.OnLog += channel.Stdout;
+            ////
             qsim.OnDisplayableDiagnostic += channel.Display;
```

Comment:
We'll need to add the logic that was used to display histograms via the shell router as an event handler here as well.

Reply:
I assume that displaying histograms is the functionality related to "div_id"/DivId. That is exactly what I would like to discuss.

Reply:
Resolved in a [repeated attempt](https://github.com/microsoft/qsharp-runtime/pull/898).